internal/codespaces/rpc/invoker.go

```diff
@@ -229,3 +234,35 @@ func (i *invoker) StartSSHServerWithOptions(ctx context.Context, options StartSS
 
 	return port, response.User, nil
 }
+
+// Periodically check whether there is a reason to keep the connection alive, and if so, notify the codespace to do so
+func (i *invoker) heartbeat(ctx context.Context, interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+
+	// Send initial connection heartbeat (no need to throw if we fail to get a response from the server)
+	_ = i.notifyCodespaceOfClientActivity(ctx, connectedEventName)
```

Comment:
Wondering if there's a debug log we could write to in order to capture errors here